astropy/io/fits/connect.py

```diff
@@ -265,21 +265,31 @@ def read_table_fits(
         # string, empty strings.
         # Since Multi-element columns with dtypes such as '2f8' have a subdtype,
         # we should look up the type of column on that.
+        # Also propagate TNULL (for ints) or the FITS default null value for
+        # floats and strings to the column's fill_value to ensure round trips
+        # preserve null values.
         masked = mask = False
         coltype = col.dtype.subdtype[0].type if col.dtype.subdtype else col.dtype.type
         if col.null is not None:
```

Comment:
`fill_value` is not initialized. In this `if / elif / elif` block there is no `else`, so `fill_value` could in theory wind up undefined (in the first column) or still holding the `fill_value` from the previous column.